psamm/datasource/sbml.py

```diff
@@ -709,11 +709,51 @@ def _add_flux_bounds(self, r_id, model, flux_limits, equation):
             upper_str = text_type(flux_limits[r_id][1])
         return lower_str, upper_str
 
+    def _add_gene_associations(self, r_id, r_genes, gene_set, r_tag):
+        if r_id in r_genes:
+            genes = ET.SubElement(
+                r_tag, _tag('geneProductAssociation', FBC_V2))
+            if isinstance(r_genes[r_id], list):
+                e = Expression(
+                    And(*(Variable(i) for i in r_genes[r_id])))
+            else:
+                e = Expression(r_genes[r_id])
+            gene_stack = [(e._root, genes)]
+            while len(gene_stack) > 0:
+                current, parent = gene_stack.pop()
+                if isinstance(current, Or):
+                    gene_tag = ET.SubElement(parent, _tag('or', FBC_V2))
+                elif isinstance(current, And):
+                    gene_tag = ET.SubElement(parent, _tag('and', FBC_V2))
+                elif isinstance(current, Variable):
+                    gene_tag = ET.SubElement(parent, _tag(
+                        'geneProductRef', FBC_V2))
+                    gene_tag.set(_tag('geneProduct', FBC_V2),
+                                 current.symbol)
```

Comment:
I don't think you can use `current.symbol` directly here. `current.symbol` is the label of the gene but this needs to be the SBML id for the gene. You seem to be generating the unique IDs in `_add_gene_list` so you need to supply that ID mapping to this method or work around it in some other way.